### PR TITLE
Update .NET Core to 2.0.0

### PR DESCRIPTION
**Pull request by mikeharder**
- Addresses https://github.com/aspnet/benchmarks/issues/276

Will update and submit PR to TechEmpower once 2.0.0 is RTM.

### CODE REVIEW

File: frameworks/CSharp/aspnetcore/Benchmarks/Benchmarks.csproj [hunk 1]
```diff
@@ -13,17 +13,17 @@
...
+    <PackageReference Include="Microsoft.AspNetCore.Server.HttpSys" Version="2.0.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Server.Kestrel" Version="2.0.0" />
+    <PackageReference Include="Microsoft.AspNetCore.StaticFiles" Version="2.0.0" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="2.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.CommandLine" Version="2.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.EnvironmentVariables" Version="2.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="2.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Options.ConfigurationExtensions" Version="2.0.0" />
+    <PackageReference Include="Npgsql" Version="3.2.5" />
+    <PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="2.0.0-preview2" />
```
Comment (dougbu) [inline frameworks/CSharp/aspnetcore/Benchmarks/Benchmarks.csproj]: The `2.0.0-preview1` bits led to a compilation error when using post-preview1 versions of EF. So, we should already be building with this EF provider in the Benchmarks repo. But, I'm not sure we've done any PostgreSQL testing with these packages and that should be the focus of any manual work on this PR.

File: frameworks/CSharp/aspnetcore/Benchmarks/Benchmarks.csproj [hunk 2]
```diff
@@ -13,17 +13,17 @@
 
   <ItemGroup>
     <PackageReference Include="Dapper" Version="1.50.2" />
```
Comment (dougbu) [inline frameworks/CSharp/aspnetcore/Benchmarks/Benchmarks.csproj]: Wow, this has been the current version for more than a year 🎂

**Comment by dougbu** [comment]
@mikeharder is this still WIP?

**Comment by mikeharder** [comment]
I'll let you know when I'm done with all the planned changes, should be later today.

**Comment by mikeharder** [comment]
@dougbu: Ready for your review, then I'll submit the PR to https://github.com/TechEmpower/FrameworkBenchmarks.

**Comment by mikeharder** [comment]
Blocked by https://github.com/npgsql/Npgsql.EntityFrameworkCore.PostgreSQL/issues/223

**Comment by dougbu** [comment]
:shipit: again

Great npgsql/Npgsql.EntityFrameworkCore.PostgreSQL#223 took less than a week to fix.

**Pull request closed without merging**